lang/test/org/partiql/lang/eval/visitors/SelectListItemAliasVisitorTransformTests.kt

```diff
@@ -0,0 +1,166 @@
+package org.partiql.lang.eval.visitors
+
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ArgumentsSource
+import org.partiql.lang.util.ArgumentsProviderBase
+
+class SelectListItemAliasVisitorTransformTests : VisitorTransformTestBase() {
+
+    class ArgsProvider : ArgumentsProviderBase() {
+        override fun getParameters(): List<Any> = listOf(
+            // singular variable references
+            TransformTestCase(
+                """
+                SELECT 
+                    a 
+                FROM foo
+                """,
+                """
+                SELECT 
+                    a AS a 
+                FROM foo
+                """
+            ),
+            TransformTestCase(
+                """
+                SELECT 
+                    a,
+                    b
+                FROM foo
+                """,
+                """
+                SELECT 
+                    a AS a,
+                    b AS b 
+                FROM foo
+                """
+            ),
+            TransformTestCase(
+                """
+                SELECT 
+                    a,
+                    B,
+                    c
+                FROM foo
+                """,
+                """
+                SELECT 
+                    a AS a,
+                    B AS B,
+                    c AS c 
+                FROM foo
+                """
+            ),
+            // path expressions
+            TransformTestCase(
+                """
+                SELECT 
+                    x.a
+                FROM foo
+                """,
+                """
+                SELECT 
+                    x.a AS a 
+                FROM foo
+                """
+            ),
+            TransformTestCase(
+                """
+                SELECT 
+                    x.a,
+                    x.B
+                FROM foo
+                """,
+                """
+                SELECT 
+                    x.a AS a,
+                    x.B AS B 
+                FROM foo
+                """
+            ),
+            TransformTestCase(
+                """
+                SELECT 
+                    x.a,
+                    x.B,
+                    x.c
+                FROM foo
+                """,
+                """
+                SELECT 
+                    x.a AS a,
+                    x.B AS B,
+                    x.c AS c 
+                FROM foo
+                """
+            ),
```

Comment:
Could add a multi-nested test. Something like:

```
TransformTestCase(
	"""
	SELECT 
	    x.foo.a,
	    x.bar.B
	FROM foo
	""",
	"""
	SELECT 
	    x.foo.a AS a,
	    x.bar.B AS B
	FROM foo
	"""
```

Reply:
I would also add a nested query. Do we deal with already aliased names in the outer query that appear in the nested query? Do we also make sure we do not accidentally capture (shadow) names?

Reply:
done